### ISSUE TITLE
unifi.sh

### ISSUE THREAD

**Issue by simonlock** [Neilpang/acme.sh#1824]
Steps to reproduce
------------------

Debug log
-----------------

```
acme.sh  --issue .....   --debug 2
```

**Comment by simonlock** [comment]
acme.sh has generated the following files

ca.cer
fullchain.cer
domainaname.cer
domainaname.conf
domainaname.csr
domainaname.csr.conf
domainaname.key

Could someone please explain how this script is used to deploy the generated certificates into the unif controller? Has anyone had success using this script?

Thanks

**Comment by 3nth** [comment]
Yes, it works for me on ubuntu with unifi controller. Perhaps your keystore file is in a different location or uses a non default password? The script default is /usr/lib/unifi/data/keystore and can be overridden by add this to your conf file for your domain:
`DEPLOY_UNIFI_KEYSTORE="/usr/lib/unifi/data/keystore"`

**Comment by medmunds** [comment]
@simonlock unifi.sh is an acme.sh "deploy hook". You don't run it directly, instead you refer to it in an acme.sh deploy command. Example:

```shell
acme.sh --deploy -d "unifi.example.com" --deploy-hook unifi
```

(Where unifi.example.com is the domain you issued a cert for with an earlier `acme.sh --issue` command.)

To use the unifi deploy hook, you must be running acme.sh locally on your Unifi Controller machine. (The unifi deploy hook directly modifies the controller's keystore file.)

**Comment by neilpang** [comment]
Closing as stale: no activity for over 18 months. If this still applies on the current release, please open a new issue with the `--debug 2` log.
